carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/decorator/ExtendedWebElement.java

```diff
@@ -318,24 +319,34 @@ public boolean isPresent(By by, long timeout) {
      */
 	private boolean waitUntil(ExpectedCondition<?> condition, long timeout) {
 		boolean result;
+		originalException = null;
 		
 		final WebDriver drv = getDriver();
 		
 		Timer.start(ACTION_NAME.WAIT);
 		wait = new WebDriverWait(drv, timeout, RETRY_TIME).ignoring(WebDriverException.class)
 				.ignoring(NoSuchSessionException.class);
+		// StaleElementReferenceException is handled by selenium ExpectedConditions in many methods
 		try {
 			LOGGER.debug("waitUntil: starting..." + getNameWithLocator());
+			LOGGER.debug("waitUntil: starting condition: " + condition.toString());
 			wait.until(condition);
 			result = true;
 			LOGGER.debug("waitUntil: finished true..." + getNameWithLocator());
-		} catch (NoSuchElementException | TimeoutException e) {
+		} catch (NoSuchElementException e) {
 			// don't write exception even in debug mode
-			LOGGER.debug("waitUntil: NoSuchElementException | TimeoutException e..." + getNameWithLocator());
+			LOGGER.debug("waitUntil: NoSuchElementException e..." + getNameWithLocator());
 			result = false;
-		} catch (Exception e) {
+			originalException = e;
+		} catch (TimeoutException e) { 
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.53.207.202/sonarqube/coding_rules#rule_key=squid%3AS1166)